### PR TITLE
Provide some options to reduce output files

### PR DESCRIPTION
**Pull request by mbhall88**
### Added
- `-S`, `--seqid` option to name the PRG sequence, which by default uses the file name.
- `-N` shortcut for max nesting
- `-L` shortcut for min match length
- `--log` to enable specifying log file should go to path. Default behaviour is now that
  log goes to stderr by default
- `--summary` option to specify that user wants a summary file, rather than the default
  behaviour where they get no choice
- `-O`, `--output-type` option to specify what output files are required. Defaults to
  all

### Changed
- Summary file not written by default

### CODE REVIEW

File: README.md
Comment (bricoletc) [inline README.md:96]: What does `If not in aligned fasta alignment_format` mean? I think should be `If not in aligned fasta format`?

File: make_prg/io_utils.py
Comment (bricoletc) [inline make_prg/io_utils.py:155]: I like this, if I understand it uses `or` shortcircuiting to give priority to seqid :+1:

File: make_prg/subcommands/from_msa.py
Comment (bricoletc) [inline make_prg/subcommands/from_msa.py:85]: Do you guys ever use summary file? We could remove it entirely potentially, or just write it out to logging, as its two lines. @leoisl

**Review by bricoletc — COMMENTED** [review]
Lovely code, I learned from the `OutputType` class, and nice tests on the class.
All the new options make sense to me, I just have a question about `--summary`, see comments.

File: make_prg/io_utils.py
Comment (mbhall88) [reply make_prg/io_utils.py:155]: Correct.

File: make_prg/subcommands/from_msa.py
Comment (mbhall88) [reply make_prg/subcommands/from_msa.py:85]: I have never used it. Hence why I changed the default behaviour. 

If Leandro never uses it either I vote just get rid of it. It's in the commit history if we ever want it back

File: README.md
Comment (mbhall88) [reply README.md]: Urgh, yeah that was worded weird. I have made it much more succinct

**Comment by mbhall88** [comment]
> Lovely code, I learned from the `OutputType` class, and nice tests on the class.
> All the new options make sense to me, I just have a question about `--summary`, see comments.

Yeah, I recently implemented something similar in `rasusa` for allowing the user to specify the compression format. I basically stole the idea from `bcftools -O` option

File: make_prg/subcommands/from_msa.py
Comment (leoisl) [reply make_prg/subcommands/from_msa.py:81]: I've never used it, I vote to remove it also

**Review by leoisl — APPROVED** [review]
All fine, I think we all vote to remove the summary file it seems too. But fine for me to merge without this change if you wish

**Comment by mbhall88** [comment]
I guess merging this will require us to bump the version to 0.2.0. Is this going to cause problems with your prototype @leoisl?

**Comment by leoisl** [comment]
Yeah, this is a bit annoying, I am sorry. We can either: bump to version 0.1.2; or just bump to 0.2.0, and treat my fork and this fork totally separate (I don't really like this option).

The issue is that we are still not sure that we will keep the dynamic updates or change to another method to update the PanRG. However, as it is on the paper, I do think we should at least push the dynamic updates to the main repo. I think I will drop all my current priorities, and make the dynamic-update make_prg stable, finishing its interfacing with gramtools, adding unit tests, etc. Then we can merge it to this repo, and therefore have a single main branch of development. The later I push this, the harder will also be to solve the conflicts when merging. What do you think? @iqbal-lab Ok to do that, or would you prefer to leave this merge for later?

**Comment by iqbal-lab** [comment]
Fine with me, I thought it was already merged

**Comment by mbhall88** [comment]
Well it sounds like the best option is to not make a release for the stuff in this PR and instead wait until you merge your fork into this main fork. Then we can release it all as v0.2.0 - a mega release!

**Pull request merged**
